subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -892,9 +892,13 @@ defined(CONFIG_MCUMGR_SMP_COMMAND_STATUS_HOOKS)
 #endif
 
 #ifndef CONFIG_IMG_ERASE_PROGRESSIVELY
-		/* erase the entire req.size all at once */
+		/* Erase the entire req.size or flash slot all at once */
 		if (action.erase) {
+#if defined(CONFIG_MCUMGR_GRP_IMG_IMAGE_SKIP_ERASED_SECTORS)
+			rc = img_mgmt_erase_pending_upload_slot();
```

Comment:
I am not sure this is correct. The erase here happens because `img_mgmt_flash_check_empty(action->area_id);` figured out that there is something in the slot, the problem is that this function will return true even for devices that do not require erase, but erase here is not to remove data (as image erase command does) but to prepare for write, in case of no-erase devices here we have actual opposite action - slowdown due to write operation that "erases" previous image.